lib/modules/platform/github/index.ts

```diff
@@ -1910,25 +1911,25 @@ export async function mergePr({
     }
   }
   if (!automerged) {
-    // We need to guess the merge method and try squash -> rebase -> merge
-    options.body.merge_method = 'rebase';
+    // We need to guess the merge method and try squash -> merge -> rebase
+    options.body.merge_method = 'squash';
     try {
       logger.debug({ options, url }, `mergePr`);
       automergeResult = await githubApi.putJson(url, options);
     } catch (err1) {
-      logger.debug({ err: err1 }, `Failed to rebase merge PR`);
+      logger.debug({ err: err1 }, `Failed to squash merge PR`);
       try {
-        options.body.merge_method = 'squash';
+        options.body.merge_method = 'merge';
         logger.debug({ options, url }, `mergePr`);
         automergeResult = await githubApi.putJson(url, options);
       } catch (err2) {
-        logger.debug({ err: err2 }, `Failed to merge squash PR`);
+        logger.debug({ err: err2 }, `Failed to merge commit PR`);
         try {
-          options.body.merge_method = 'merge';
+          options.body.merge_method = 'rebase';
           logger.debug({ options, url }, `mergePr`);
           automergeResult = await githubApi.putJson(url, options);
         } catch (err3) {
-          logger.debug({ err: err3 }, `Failed to merge commit PR`);
+          logger.debug({ err: err3 }, `Failed to rebase merge PR`);
           logger.info({ pr: prNo }, 'All merge attempts failed');
           return false;
```

Comment:
I took the liberty of changing this bit too when I realized the comment was no longer in sync with the code. It now follow the same order (squash -> merge -> rebase)